### PR TITLE
Extension Settings doc.md

### PR DESCRIPTION
**Pull request by adam-fowler**
Document giving more detail what each setting does

### CODE REVIEW

File: docs/settings.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Extension Settings
+
+The Visual Studio Code Swift extension comes with a number of settings you can use to control how it works. This document details what each of these settings does.
+
+- **Swift: Path**
+
+This is the folder that the `swift` executable can be found in. If this is not set then the extension will look for executables in the $PATH. The extension will also use this setting when looking for other executables it requires like `sourcekit-lsp` and `lldb`.
```
Comment (0xTim) [inline docs/settings.md:7]:
```suggestion
This is the folder that the `swift` executable can be found in. If this is not set then the extension will look for executables in the `$PATH`. The extension will also use this setting when looking for other executables it requires like `sourcekit-lsp` and `lldb`.
```
```suggestion
This is the folder that the `swift` executable can be found in. If this is not set then the extension will look for executables in the $PATH. The extension will also use this setting when looking for other executables it requires like `sourcekit-lsp` and `lldb`.
```

File: docs/settings.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+The path of the `sourcekit-lsp` executable. As mentioned above the default is to look in the folder where the `swift` executable is found.
+
+- **Sourcekit-LSP: Server Arguments**
+
+This is a list of arguments that will be passed to the SourceKit-LSP. Argument keys and values should be provided as separate entries in the array e.g. `['--log-level', 'debug']`.
+
+- **Sourcekit-LSP: Inlay Hints**
+
+This controls the display of Inlay Hints. Inlay Hints are variable annotations indicating their inferred type. They are only available if you are using Swift 5.6 or later.
```
Comment (0xTim) [inline docs/settings.md:43]: Is this enabled by default with 5.6?

Comment (0xTim) [inline docs/settings.md:43]: Is this enabled by default with 5.6?

**Review by 0xTim — APPROVED** [review]
Couple of queries/very minor things

File: docs/settings.md (hunk 2 above)
Comment (adam-fowler) [reply docs/settings.md:43]: Yes

Comment (0xTim) [reply docs/settings.md:43]: Might be worth noting that?

**Comment by adam-fowler** [comment]
@compnerd , @stevapple I added details about the advanced settings here (runtime path and sdk). Don't know if you guys have anything to add. SDK description is a little sparse.

**Comment by stevapple** [comment]
@adam-fowler I will add the description after #266 gets in. This would distinguish host and target SDK, which are often confused and documents rarely touch their differences.

Personally I wish to see cross-compilation being enabled with 0.5.0 release, which is left the last two pieces of puzzle: custom `SDKROOT` (#266) and custom target (yet to implement but I think it should be quick).

**Comment by adam-fowler** [comment]
> @adam-fowler I will add the description after #266 gets in. This would distinguish host and target SDK, which are often confused and documents rarely touch their differences.
> 
> Personally I wish to see cross-compilation being enabled with 0.5.0 release, which is left the last two pieces of puzzle: custom `SDKROOT` (#266) and custom target (yet to implement but I think it should be quick).

I think I might delay that to a v0.5.1 or v0.6.0. There are a lot of fixes and improvements I would like to get out and once we get swift lldb working I am planning to do a release then.

**Comment by stevapple** [comment]
> I think I might delay that to a v0.5.1 or v0.6.0. There are a lot of fixes and improvements I would like to get out and once we get swift lldb working I am planning to do a release then.

At least we should have #266 in first. Confusing the host and target SDK is not so good (and it adds to the cost of teaching the correct), so we should avoid making such confusion if possible.

**Pull request merged**